master/internal/api_experiment.go

```diff
@@ -150,3 +150,43 @@ func (a *apiServer) PauseExperiment(
 		return resp, nil
 	}
 }
+
+func (a *apiServer) CancelExperiment(
+	ctx context.Context, req *apiv1.CancelExperimentRequest,
+) (
+	resp *apiv1.CancelExperimentResponse, err error) {
+	ok, err := a.m.db.CheckExperimentExists(int(req.Id))
+	switch {
+	case err != nil:
+		return nil, status.Errorf(codes.Internal, "failed to check if experiment exists: %s", err)
+	case !ok:
+		return nil, status.Errorf(codes.NotFound, "experiment %d not found", req.Id)
+	}
+
+	addr := actor.Addr("experiments", req.Id).String()
+	err = a.actorRequest(addr, req, &resp)
+	if status.Code(err) == codes.NotFound {
+		return &apiv1.CancelExperimentResponse{}, nil
```

Comment:
correct because at this point if actor doesn't find it it means that this experiment exists (in DB) but is in a terminal state. how does that sound to you

Reply:
yeah that sounds perfect to me, easiest behavior for a user to design stuff with imo.